app/lib/glean.rb

```diff
@@ -0,0 +1,178 @@
+  end
+end
+
+module Glean
+  class GleanEventsLogger
+    def initialize(app_id:, app_display_version:, app_channel:, logger_options:)
+      @app_id = app_id # string - Application Id to identify application per Glean standards
+      @app_display_version = app_display_version # string - Version of application emitting the event
+      @app_channel = app_channel # string - Application Id to identify application per Glean standards
+      @logger = Logger.new(logger_options)
+
+      # Logger configuration
+      @logger.formatter = proc do |severity, datetime, _progname, msg|
+        date_format = datetime.to_i
+        logger_name = 'glean'
+        "#{JSON.dump(Timestamp: date_format.to_s, Logger: logger_name.to_s, Type: GLEAN_EVENT_MOZLOG_TYPE.to_s, Severity: severity.ljust(5).to_s, Pid: Process.pid.to_s, Fields: msg)}\n"
+      end
+
+      # Generated events
+      # Event triggered by the backend to record the change in state of an object (e.g. API requests to the mozilla.social Mastodon server). In the future, we could potentially use this event to track changes in state to core Mastodon objects (e.g. accounts and posts).
+      @backend_object_update = BackendObjectUpdateEvent.new(self)
+    end
+
+    def _record(
+      # The Adjust device ID for this user, if available.
+      identifiers_adjust_device_id:,
+      # The user's FxA account ID, if available.
+      identifiers_fxa_account_id:,
+      # The user's full account handle, with domain. For example, `account_name@mozilla.social`.
+      identifiers_mastodon_account_handle:,
+      # The user's numeric account ID from Mastodon.
+      identifiers_mastodon_account_id:,
+      # The device user agent string.
+      identifiers_user_agent:,
+      # full user_agent value from controller context
+      user_agent:,
+      # ip address value from controller context
+      ip_address:,
+      # event being sent in the ping
+      event:
+    )
+      t_utc = Time.now.utc
+      # create raw metrics hash that can have nil values
+      metrics_raw = {
+        'string' => {
+          'identifiers.adjust_device_id' => identifiers_adjust_device_id,
+          'identifiers.fxa_account_id' => identifiers_fxa_account_id,
+          'identifiers.mastodon_account_handle' => identifiers_mastodon_account_handle,
+          'identifiers.mastodon_account_id' => identifiers_mastodon_account_id,
+          'identifiers.user_agent' => identifiers_user_agent,
+        },
+      }
+      # filter out key value pairs where value is nil
+      metrics_raw.each do |key, value|
+        metrics_raw[key] = value.compact.transform_values(&:to_s)
+      end
+      # filter out metrics with empty hashes
+      metrics = metrics_raw.reject { |_k, v| v.empty? }
+      event_payload = {
+        # `Unknown` fields below are required in the Glean schema, however they are not useful in server context.
+        'client_info' => {
+          'telemetry_sdk_build' => 'glean_parser v0.1.dev1004+g4822435',
```

Comment:
This is a bit of a misnomer since you don't really use an SDK in the server context. Since `telemetry_sdk_build` is a required field in ping schema we decided to put there a version of the parser that's used to generate this code.

Reply:
since this value is an artifact of the parser and not the dependencies of mastodon, I think we are in agreement that this change is probably not needed.